src/api/v1/utils.ts

```diff
@@ -5,6 +5,28 @@ import * as Hapi from 'hapi';
 import * as Boom from 'boom';
 import { ApiResponse } from './schema/response';
 
+export interface JoiBoomError extends Boom {
+  details: {message: string}[];
```

Comment:
If this is a declaration that should actually be in `@types/boom` but is missing, we should use declaration merging instead of creating a new type.

Something like: 
```ts
declare module 'boom' {
  class Boom extends Error {
    details: { message: string }[];
  }
}
```
I don't know if you actually need to attach it to the class or to the namespace though. Take a look at `@types/boom/index.d.ts`.

Also, because I'd like to eventually replace all our custom declaration fixes with PRs to the real typings, it'd be nice to keep them in one place (which I haven't been doing)